### PR TITLE
APM SDKs

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

This PR updates our documentation to align with consistent terminology around the APM SDK, formerly known as the Datadog tracing libraries or tracers. The updates includes a new guide explaining the terminology and replacing instances of "tracer" and "tracing library" with "SDK".

- [x] TODO: Draft guide detailing APM SDK terminology. [Preview](https://docs-staging.datadoghq.com/brett0000FF/dd-sdks/tracing/guide/sdks/)
- [x] TODO: Replace `tracer` and `tracing library` with SDK (or API) throughout all APM docs. **Note**: Avoiding code samples or where content refers specifically to `Tracer` in code example.
- [x] TODO: Make separate PR for Ruby changes in dd-trace-rb. [Link](https://github.com/DataDog/dd-trace-rb/pull/3819)
- [ ] TODO: Make separate PR for docs outside /tracing/ like ASM, OTel, etc.
- [ ] TODO: After initial work is complete, add Vale rule.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Use the dependency map to visualize service-to-service communication and gain in
 
 To opt in, Datadog recommends you adjust your:
 - [Datadog Agent](#datadog-agent-configuration) (or [OpenTelemetry collector](#opentelemetry-collector)) configuration
-- [APM tracing libraries](#apm-tracing-libary-configuration) configuration
+- [APM APM SDKs](#apm-tracing-libary-configuration) configuration
```
Comment (zacharycmontoya) [inline content/en/tracing/guide/inferred-service-opt-in.md:46]:
```suggestion
- [APM SDKs](#apm-tracing-libary-configuration) configuration
```

**Review by cswatt — COMMENTED** [review]
Some feedback! Note that my main note on the SDK guide is very much just an opinion and can be ignored.

File: content/en/tracing/guide/sdks.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - link: 'tracing/trace_collection/automatic_instrumentation/dd_libraries/'
+    tag: 'Documentation'
+    text: 'Add the Datadog APM SDK'
+algolia:
+  tags: ['tracer', 'tracing library']
+---
+
+## Overview
+
+Datadog now consistently uses the term **APM SDK** to refer to Datadog Application Performance Monitoring tools, previously known as **tracing libraries** or **tracers**.
```
[the remark this answers is not in the document]
Comment (brett0000FF) [reply content/en/tracing/guide/sdks.md]: @cswatt  I appreciate your perspective! This is a somewhat unusual page for us to write, and your feedback is really useful. I slightly restructured and rewrote the content to put more emphasis on how this affects and relates to our users. The PM appreciated some of this content, so I didn't want to remove it entirely.

Please let me know what you think--thanks!

**Review by cswatt — APPROVED** [review]

File: content/en/tracing/trace_collection/compatibility/go.md [hunk 3]
```diff
@@ -17,18 +17,18 @@ further_reading:
 
 The Go Datadog Trace library is open source. View the [GitHub repository][1] for more information.
 
-The Go Datadog Trace Library has a [version support policy][2] defined for Go versions. The two latest releases of Go are fully supported, while the third newest release is considered in [maintenance][3]. Older versions may function, but no support is provided by default. For special requests, [contact support][4]. 
+The Go Datadog Trace Library has a [version support policy][2] defined for Go versions. The two latest releases of Go are fully supported, while the third newest release is considered in [maintenance][3]. Older versions may function, but no support is provided by default. For special requests, [contact support][4].
```
Comment (bm1549) [inline content/en/tracing/trace_collection/compatibility/go.md:20]: Seems like we missed a string - `Go Datadog Trace Library` should be `Go APM SDK`

```suggestion
The Go APM SDK has a [version support policy][2] defined for Go versions. The two latest releases of Go are fully supported, while the third newest release is considered in [maintenance][3]. Older versions may function, but no support is provided by default. For special requests, [contact support][4].
```

File: content/en/tracing/trace_collection/compatibility/dotnet-framework.md [hunk 4]
```diff
@@ -95,6 +95,22 @@ The [latest version of the .NET Tracer][5] can automatically instrument the foll
 
 Don't see the library you're looking for? First, check if the library produces observability data compatible with OpenTelemetry (see [Using OpenTelemetry Instrumentation Libraries][11] for more details). If not, Datadog is continually adding additional support. [Check with the Datadog team][6] for help.
 
+## OpenTelemetry based integrations
```
Comment (bm1549) [inline content/en/tracing/trace_collection/compatibility/dotnet-framework.md:98]: I didn't expect to see new content here - is this part of a bad merge?

File: content/en/tracing/trace_collection/compatibility/dotnet-core.md [hunk 5]
```diff
@@ -92,9 +92,25 @@ The [latest version of the .NET Tracer][4] can automatically instrument the foll
 
 Don't see the library you're looking for? First, check if the library produces observability data compatible with OpenTelemetry (see [Using OpenTelemetry Instrumentation Libraries][13] for more details). If not, Datadog is continually adding additional support. [Check with the Datadog team][5] for help.
 
+## OpenTelemetry based integrations
```
Comment (bm1549) [inline content/en/tracing/trace_collection/compatibility/dotnet-core.md:95]: Didn't expect to see this either

**Review by bm1549 — COMMENTED** [review]
LGTM aside from the comments on Go and .NET

File: content/en/tracing/trace_collection/compatibility/dotnet-framework.md (hunk 4 above)
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/compatibility/dotnet-framework.md:98]: This must be a bad merge. This content was removed in a recent PR so it looks like added content. @brett0000FF please remove this section entirely

File: content/en/tracing/trace_collection/compatibility/dotnet-core.md (hunk 5 above)
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/compatibility/dotnet-core.md:95]: This must be a bad merge. This content was removed in a recent PR so it looks like added content. @brett0000FF please remove this section entirely

File: config/_default/menus/main.en.yaml [hunk 6]
```diff
@@ -5125,7 +5125,7 @@ menu:
       parent: appsec_enabling_single_step
       identifier: appsec_enabling_single_step_code
       weight: 2102
-    - name: Using Datadog Tracing libraries
+    - name: Using Datadog Tracing Libraries
```
Comment (lucaspimentel) [inline config/_default/menus/main.en.yaml:5128]: We don't want to change this one?
```suggestion
    - name: Using Datadog APM SDKs
```

File: content/en/tracing/trace_collection/compatibility/dotnet-framework.md (hunk 4 above)
Comment (brett0000FF) [reply content/en/tracing/trace_collection/compatibility/dotnet-framework.md]: Good catch--thanks! Removed these sections.

File: config/_default/menus/main.en.yaml (hunk 6 above)
Comment (brett0000FF) [reply config/_default/menus/main.en.yaml]: Thanks for catching. This should be fixed now.

**Comment by brett0000FF** [comment]
Pausing this effort for now ⏸️

**Pull request closed without merging**